File: src/main/java/com/javahere/reachyourgoal/entity/Task.java
package com.javahere.reachyourgoal.entity;

import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.util.Set;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
public class Task {
    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE)
    private Long id;

    @NotBlank
    private String name;

    @Size(max = 400)
    private String description;

    @NotNull
    private LocalDate taskDate;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @Enumerated(EnumType.STRING)
    private TaskStatus taskStatus = TaskStatus.NOT_STARTED;

    @ManyToMany
    private Set<Attachment> attachments;
}

File: src/main/kotlin/me/javahere/reachyourgoal/domain/Task.kt
package me.javahere.reachyourgoal.domain

import me.javahere.reachyourgoal.dto.TaskDto
import me.javahere.reachyourgoal.util.Transformable
import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import org.springframework.data.relational.core.mapping.Table
import java.util.UUID

@Table(name = "tasks")
class Task(
    @Id
    @Column("id")
    val id: UUID? = null,
    @Column("name")
    val name: String,
    @Column("description")
    val description: String? = null,
    @Column("spent_time")
    val spentTime: Long = 0L,
    @Column("status")
    val status: TaskStatus = TaskStatus.NOT_STARTED,
    @Column("user_id")
    val userId: UUID,
) : Transformable<TaskDto> {
    override fun transform(): TaskDto {
        return TaskDto(
            id = id!!,
            name = name,
            description = description,
            spentTime = spentTime,
            status = status,
            userId = userId,
        )
    }
}
